manta-pay/abstract-protocol.tex

```diff
@@ -190,133 +176,169 @@ \subsection{Abstract Cryptographic Schemes}
     \begin{align*}
-        S.\sk                              &:= \sk \\
-        S.\rk.\spend                       &:= \KA.\derive(S.\sk.\spend) \\
-        \Some\left(S.\esk, S.\asset\right) &:= \HPKE.\decrypt(S.\sk.\view, \epk, C_\note) \\
-        S.\cm                              &:= \COM^{\UTXO}_{S.\esk}(S.\rk.\spend, S.\asset) \\
-        \Some\left(S.\cm_z, S.\cm_w\right) &:= \UTXOSet.\contains(S.\cm, \Ledger.\utxos()) \\
-        S.\vn                              &:= \COM^{\VN}_{S.\sk.\spend}(S.\cm)
+        t      &:= \iszero(\sa.\ID) \,\wedge\, \iszero(\sa.\VALUE) \\
+        \asset &:= \select(t, \sa, \pa) \\
+        \cm    &:= \COM^{\UTXO}(r, \pk, \sa) \\
+        \utxo  &:= (t, \pa, \cm) \\
+        h      &:= H^{\UTXO}(\utxo) \\
+        \Some\left(h_z, h_w\right) &:= \UTXOSet.\contains(h, \Ledger.\utxos()) \\
+        n      &:= \COM^{\textsf{N}}(\ak, h) \\
+        \epk   &:= g \cdot \esk \\
+        \Cout  &:= \AUTHout.\encrypt(\pk \cdot \esk, \select(t, \sa, \pa))
     \end{align*}
 \end{definition}
 
 \begin{definition}[\Transfer{} Sender Post]
     A \SenderPost{} is the following tuple extracted from a \Sender{}:
     \begin{align*}
-        \cm_z &: \UTXOSet.\Output \\
-        \vn   &: \VoidNumber
+        h_z   &: \UTXOSet.\Output \\
+        n     &: \Nullifier \\
+        \epk  &: \G \\
+        \Cout &: \AUTHout.\Ciphertext
     \end{align*}
     which are the parts of a \Sender{} which should be \emph{posted} to the \Ledger{}.
 \end{definition}
 
 \begin{definition}[\Transfer{} Receiver]
     A \Receiver{} is the following tuple:
     \begin{align*}
-        \rk       &: \ReceivingKey \\
-        \esk      &: \KA.\SecretKey \\
-        \asset    &: \Asset \\
-        \cm       &: \UTXO \\
-        \note     &: \EncryptedNote
+        \pk    &: \zkAddress \\
+        r      &: \F \\
+        \sa    &: \Asset \\
+        \pa    &: \Asset \\
+        t      &: \Bool \\
+        \asset &: \Asset \\
+        \cm    &: \F \\
+        \utxo  &: \UTXO \\
+        \esk   &: \Z_p \\
+        \epk   &: \G \\
+        \Cin   &: \AUTHin.\Ciphertext
     \end{align*}
     
-    A \Receiver{}, $R$, is constructed from a receving key $\rk : \ReceivingKey$, an asset $\asset : \Asset$, and a random ephemeral secret key $\esk : \HPKE.\KA.\SecretKey$ with the following algorithm:
+    A \Receiver{}, $R$, is constructed in the following way:
     \begin{align*}
-        R.\rk       &:= \rk \\
-        R.\esk      &:= \esk \\
-        R.\asset    &:= \asset \\
-        R.\cm       &:= \COM^{\UTXO}_{R.\esk}(R.\rk.\spend, R.\asset) \\
-        R.\note     &:= \HPKE.\encrypt\left(R.\rk.\view, R.\esk, (R.\esk, R.\asset)\right)
+        t      &:= \iszero(\sa.\ID) \,\wedge\, \iszero(\sa.\VALUE) \\
+        \asset &:= \select(t, \sa, \pa) \\
+        \cm    &:= \COM^{\UTXO}(r, \pk, \sa) \\
+        \utxo  &:= (t, \pa, \cm) \\
+        \epk   &:= g \cdot \esk \\
+        \Cin   &:= \AUTHin.\encrypt(\pk \cdot \esk, (r, \sa))
```

Comment:
Not true, you start the construction of a receiver assuming they have access to the Asset, but for that they need to decrypt the corresponding Encrypted Note.

Reply:
The decryption of the note is not part of these types, that is part of the synchronization protocol which is not specified here.